### DISCUSSION TITLE
Automation Button

### DISCUSSION THREAD

**Discussion by Churator** [Blackymas/NSPanel_HA_Blueprint#331] [General]
`I've` created automation that send position to my roller shutter. Running the automation manually works exactly as expected.
When added to panel blueprint, I see the buttons but they do nothing, just light up when I click and no light when I client again. Behave like button while its not really a button.

Tried to see if automation should be configured differently, couldn't find anything.

![image](https://user-images.githubusercontent.com/62523824/218328275-8ae56124-1476-4979-869b-f5506654097e.png)

and blueprint settings

![image](https://user-images.githubusercontent.com/62523824/218328300-ea164672-b09f-4d03-87f3-b1abf4810a1e.png)

**Comment by deejaybeam** [comment]
What I see here is that you have assigned an automation to a button.
Now you expect that a push on the button executes the automation - but thats wrong.
What you want is to use a script - if you a assign a script, this will be executed.
See also: https://github.com/Blackymas/NSPanel_HA_Blueprint/issues/320
A script does the same as an automation, but it has no trigger like an automation - what in your case also makes no sense, because you have not assigned a trigger.
So, you just have to use a script for that.
The use case to enable/disable an automation is e.g. if you have an (running - triggered from somewhere) automation like heating-control, holiday-mode,... and with the button you are able to turn the automation on/off (enable/disable)

**Reply by Churator** [reply]
Totally make sense, wish it was documented for clarification.

**Comment by edwardtfn** [comment]
Let's discuss about this on https://github.com/Blackymas/NSPanel_HA_Blueprint/discussions/810#discussioncomment-7596539

**Comment by edwardtfn** [comment]
I've added a selector to switch between **Toggle** and **Trigger** for button automations. This feature is now in the `dev` branch and will be included in our next release.
